File: ClientApp/src/services/userService.ts
import User from "../models/user";

const API_BASE_URL = "https://localhost:7113";

export const getUsers = async (): Promise<User[]> => {
  const response = await fetch(`${API_BASE_URL}/users`);
  if (response.status === 204) {
    console.log("no data");
    return [];
  } else {
    const data = await response.json();
    return data;
  }
};

File: Services/UserService.cs
using InvoicesManagement.Models;

namespace InvoicesManagement.Services
{
    public class UserService
    {
        private readonly GraphQLService _graphQLService;

        public UserService(GraphQLService graphQLService)
        {
            _graphQLService = graphQLService;
        }
        public async Task<List<User>> GetUsers()
        {
            var query = @"
                query GetUsers {
                    Users {
                        id
                        name
                        email
                        phone
                        address
                    }
                }
            ";

            var response = await _graphQLService.SendGraphQlRequest(query);
            return _graphQLService.ParseGraphQlResponse<List<User>>(response, "Users");
        }
    }
}
